src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -4231,7 +4231,7 @@ public bool ConnectAsync(SocketAsyncEventArgs e)
                     {
                         InternalBind(new IPEndPoint(IPAddress.Any, 0));
                     }
-                    else
+                    else if (endPointSnapshot.AddressFamily == AddressFamily.InterNetworkV6)
```

Comment:
Please explain how this is breaking app-compat with Desktop. 

This is needed for sockets that don't need local IP address. The only one now is Unix Domain Socket, and it's tested by the unit tests, the success scenario in particular. How can it be tested on Desktop where there cannot be Unix?